### PR TITLE
refactor(rector): Run rector php74 upgrade

### PR DESCRIPTION
**Pull request by olivernybroe**
| Q             | A
| ------------- | ---
| Bug fix?      | no
| New feature?  | yes

This PR upgrades our code to php74 with an auto upgrade from rector.

I have fixed some errors introduced from this auto upgrade and everything should be working now. 😄 

This might be a hard one to look through as there are soo many small changes, sorry! 👍

### CODE REVIEW

**Comment by nunomaduro** [comment]
@olivernybroe Maybe this config:

```
# rector.yaml
parameters:
    sets:
        - 'action-injection-to-constructor-injection'
        - 'array-str-functions-to-static-call'
        - 'celebrity'
        - 'doctrine'
        - 'phpstan'
        - 'phpunit-code-quality'
        - 'solid'
        - 'early-return'
        - 'doctrine-code-quality'
        - 'dead-code'
        - 'code-quality'
        - 'php71'
        - 'php72'
        - 'php73'
        - 'php74'
```

**Comment by nunomaduro** [comment]
Can you also add rector as part of the `composer lint command`?

**Comment by olivernybroe** [comment]
@nunomaduro Cool, I'll try running with those sets and add it to composer command! :)

**Comment by TomasVotruba** [comment]
Let me know if you need any help ;)

**Comment by olivernybroe** [comment]
@TomasVotruba Thanks! 👍 Think I fixed all the errors in the code which caused problems.

The biggest problem I had was actually that rector removed a bunch of newlines between methods 😢 But phpcs fixed that.

**Comment by nunomaduro** [comment]
@olivernybroe Can we apply `@noRector` on the file we copied from phploc? The one that contains parses all the metrics?

**Comment by olivernybroe** [comment]
Alright this should be ready now 👍

**Comment by TomasVotruba** [comment]
@olivernybroe Good job with using it!

Yes, AST cannot handle spaces, only coding standards, see https://github.com/rectorphp/rector/blob/master/README.md#how-to-apply-coding-standards

Were there some false positives that should be fixed in core?

File: rector.yaml [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+        - 'phpunit-code-quality'
+        - 'solid'
+        - 'early-return'
+        - 'doctrine-code-quality'
+        - 'dead-code'
+        - 'code-quality'
+        - 'php71'
+        - 'php72'
+        - 'php73'
+        - 'php74'
```
Comment (TomasVotruba) [inline rector.yaml:27]: Also I recommend `privatization` set
https://www.tomasvotruba.com/blog/2020/03/30/dont-show-your-privates-to-public

File: src/Application/Adapters/Laravel/Commands/InsightsCommand.php [hunk 2]
```diff
@@ -12,17 +12,16 @@
...
+use RuntimeException;
 use Symfony\Component\Console\Output\ConsoleOutputInterface;
 
 /**
  * @internal
  */
 final class InsightsCommand extends Command
 {
-    /** @var string */
     protected $name = 'insights';
```
Comment (TomasVotruba) [inline src/Application/Adapters/Laravel/Commands/InsightsCommand.php:23]: Is this false positive or blocked by parent?

File: src/Application/Console/Formatters/Console.php [hunk 3]
```diff
@@ -45,34 +46,56 @@ final class Console implements Formatter
...
+    <%style_color%>%block_size%</>
+    EOD;
+
+    private const CATEGORY_COLOR = [
+        'Code' => 'cyan',
+        'Complexity' => 'green',
+        'Architecture' => 'blue',
+        'Style' => 'yellow',
+        'Security' => 'red',
+    ];
```
Comment (TomasVotruba) [inline src/Application/Console/Formatters/Console.php:81]: Love these constants :D

File: src/Application/Adapters/Laravel/Commands/InsightsCommand.php (hunk 2 above)
Comment (olivernybroe) [reply src/Application/Adapters/Laravel/Commands/InsightsCommand.php:23]: Blocked by parent 👍

File: src/Application/Console/Formatters/Console.php (hunk 3 above)
Comment (olivernybroe) [reply src/Application/Console/Formatters/Console.php:81]: Yeah, actually really helpful that they are now defined up here instead 👍

File: rector.yaml (hunk 1 above)
Comment (olivernybroe) [reply rector.yaml:27]: Oh nice, totally something for us, thanks 😄

Just right up @nunomaduro alley.

**Comment by olivernybroe** [comment]
@TomasVotruba Ah right, makes sense as the lexer ofc. ignores white spaces.

Hmm there were one of the `@noRector` I think makes sense looking into. The other one was simply a library we used that does not define their type correctly. But looking at `src/Domain/Kernel.php` we have a constant called `VERSION`, it want' us to delete that one even though it actually is used in `bin/phpinsights`

I think this has something to do with `rector` prob. not scanning the file `phpinsights` as it does not have the file extension `.php`.

**Comment by TomasVotruba** [comment]
I see. That's actually correct, you need to include  to `bin` to the paths, so Rector knows it's used.

**Comment by olivernybroe** [comment]
@TomasVotruba It is already included in paths 🤔

**Pull request merged**